documentation/modules/ROOT/pages/connectors/oracle.adoc

```diff
@@ -1355,6 +1355,11 @@ The following configuration properties are _required_ unless a default value is
 |
 |Name of the database to connect to. Must be the CDB name when working with the CDB + PDB model.
 
+||[[oracle-property-database-schema]]<<oracle-property-database-schema, `database.schema`>>
```

Comment:
There is and to some degree they function somewhat similarly if I'm being honest.

This particular option was added by the original import from Andrey and it's used when querying the log mining session view in an attempt to reduce the number of rows that get returned by the query.  Essentially the streaming step will only return rows that occurred against tables that are owned by this `database.schema` option.

If we wanted to use strictly include/exclude lists like other connector implementations, which I agree long-term we definitely should be doing and not using this adhoc configuration option, we would need to:

1. Convert these lists of regex into some formal SQL predicate
2. Stream all results from the log mining view regardless of schema to apply the include/exclude in Java.

We should definitely focus on (1) rather than (2) as the latter will lead to unnecessary performance concerns in high volume transaction environments.  Looking at Oracle documentation, there is `REGEX_LIKE(<column>, <regex>)` which allows passing a regex and a column name but I'm not entirely sure the regex support matches that of Java's.

Since I'm not 100% clear on the compatibility between Java Regex and `REGEX_LIKE`, I suggest if we want to tackle (1) that we leave `database.schema` for now, log a warning if we detect its being used and suggest that users use the schema include/exclude list choices.  We would continue using `database.schema` if supplied, otherwise we'd use the schema include/exclude lists.

What's your thoughts?  We could try & get this into 1.5 this week so we have some early testing & feedback if wanted.  It's a really easy fix after reviewing the code, shouldn't take more than an hour to change the code & run the tests.

Reply:
Couldn't we "resolve" our filter config, so to obtain the literal list of matching schemas, and then pass those name(s) to the LogMiner query? I.e. the behavior would be that of this option, only that the schema names would be sourced from our existing filter config.